compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -12,6 +12,47 @@ use rustc_span::edit_distance::find_best_match_for_name;
 use rustc_span::symbol::{sym, Symbol};
 use rustc_span::BytePos;
 
+const MAX_CHECK_CFG_NAMES_OR_VALUES: usize = 35;
+
+fn check_cfg_expected_note(
+    possibilities: &[Symbol],
+    type_: &str,
+    name: Option<Symbol>,
+    suffix: &str,
+) -> String {
+    use std::fmt::Write;
+
+    // as an anti-regression test `tests/ui/check-cfg/well-known-values.rs`
+    // needs to have the full list of expected values, so we use this dummy env
+    let n_possibilities = std::cmp::min(
+        possibilities.len(),
+        if std::env::var("RUSTC_PRINT_FULL_CHECK_CFG").is_ok() {
```

Comment:
This is adding an insta-stable environment variable to the compiler's interface.

Is there any reason this cannot be accomplished via a `-Z` flag instead?

Reply:
It's a internal-only env, like `RUSTC_LOG`, it's not supposed to by used outside of check-cfg tests.

Having a `-Z` feels overkill, since it will only be used in 1 place.

I can probably gate the env behind nightly if you want this.

Reply:
I checked in the T-compiler meeting today. ([Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-02-29/near/424038520)).

Please change this into a -Z flag. We don't want an environment variable for this sort of thing, even if its only used in one test (at the moment).

Reply:
Alright, no problem. Any preference on the name?

Maybe something like `-Zprint-full-check-cfg-suggestions`? To follow the [current `-Zprint-*`](https://github.com/rust-lang/rust/blob/71a7b66f20c551f640f2f382bc7e7923ba0a5dab/compiler/rustc_session/src/options.rs#L1798-L1814).

Reply:
I ended up going for `-Zcheck-cfg-all-expected` since all the `-Zprint-*` flags actually always print something which isn't the case for this flag.